EVAL_POWRMS: IIO and screen for rev-B:

rbudai98 opened the pull request:
Eval-powrms is an evaluation kit built around ADL5920 which is an ultrawideband, bidirectional detector that simultaneously measures forward and reverse rms power levels in a signal path, along with the return loss. The firmware allows user to interact
with setting and calibration via IIO or buttons on the PCB and can produce result both via embedded display or IIO.

## Pull Request Description

Please replace this with a detailed description and motivation of the changes. 
You can tick the checkboxes below with an 'x' between square brackets or just check them after publishing the PR. 
If this PR contains a breaking change, list dependent PRs and try to push all related PRs at the same time.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have complied with the [Submission Checklist](http://analogdevicesinc.github.io/no-OS/contributing.html#submission-checklist)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [x] I have updated the documentation (wiki pages, ReadMe etc), if applies

rbudai98 committed:
projects: eval-powrms: add firmware

Eval-powrms is an evaluation kit built around ADL5920 which
is an ultrawideband, bidirectional detector that simultaneously
measures forward and reverse rms power levels in a signal path,
along with the return loss. The firmware allows user to interact
with setting and calibration via IIO or buttons on the PCB and
can produce result both via embedded display or IIO.

Signed-off-by: rbudai98 <robert.budai@analog.com>

buha commented on drivers/adc/ad7091r5/ad7091r5.c:
```diff
@@ -57,7 +57,7 @@ int32_t ad7091r5_i2c_reg_read(struct ad7091r5_dev *dev,
 		return -EINVAL;
 
-	ret = no_os_i2c_write(dev->i2c_desc, &reg_addr, 1, 1);
+	ret = no_os_i2c_write(dev->i2c_desc, &reg_addr, 1, 0);
```
put the driver changes in a separate commit

buha commented:
the rest looks good to me but run astyle and fix the documentation

rbudai98 commented on projects/eval-powrms/src/examples/example/m24512.c:
```diff
@@ -0,0 +1,580 @@
+static int m24512_write_raw(struct m24512_dev *dev, uint16_t addr,
+                            const uint8_t *data, uint16_t len);
+
+/******************************************************************************/
```
I will solve this in a different PR, as soon this driver will be accepted, I will drop these files from this project

CiprianRegus commented on drivers/platform/maxim/max32662/maxim_delay.c:
```diff
@@ -34,6 +34,7 @@
 #include "no_os_util.h"
 #include "mxc_delay.h"
 #include "mxc_sys.h"
+#include "lvgl.h"
```
This doesn't look good. You're making the maxim delay driver dependent on an external library. I think the proper way of implementing this is to allow the user to register a function which will be called in the systick handler.

rbudai98 replied:
these are just some preliminary solutions, just for a working project... anything outside the project file will be dropped upon converting this into PR

amiclaus reviewed (commented):
please squash the commits, since they are only enhancement steps for the project that you add.

amiclaus commented on projects/eval-powrms/src/examples/example/powrms_gpios.c:
```diff
@@ -0,0 +1,618 @@
+	uint8_t pin_state;
+	ret = no_os_gpio_get_value(btn_on_off_desc, &pin_state);
+	if (ret) {
+		return ret; // Error reading pin state
```
leak?

amiclaus commented on projects/eval-powrms/src/examples/example/powrms_data_processing.c:
```diff
@@ -0,0 +1,637 @@
+	if (freq_index < FREQUENCY_RANGE_NR - 1) {
+		// Calculate temperature coefficients using floating-point arithmetic
+		for (int i = 0; i < TEMPERATURE_CORRECTION_COEFFS * 2; i++)
+			temp_coeffs[i] = (float)temperature_precision_values[freq_index *
```
When freq_index = 7 and i >= 3 -> 💣

amiclaus commented on projects/eval-powrms/src/common/common_data.c:
```diff
@@ -0,0 +1,172 @@
+struct no_os_gpio_desc *en_latch_gpio_desc;
+
+// Menu navigation related variables
+static struct no_os_gpio_desc *btn_next_desc;
```
are these ever used?

amiclaus reviewed (commented):
some comments on my side.

amiclaus reviewed (commented):
for the build issue, try skipping the wdt wrapper added in the newer versions of msdk and use directly: https://github.com/analogdevicesinc/msdk/blob/d1b7fa5e845efb5276abf60a9d94b23fc30077ac/Libraries/PeriphDrivers/Source/WDT/wdt_revb.c#L133

rbudai98 committed:
drivers: ssd1306: make i2c addr and rotation generic

Due to different available screens with same ssd1306 controler, different
i2c addresses have to be able to be defined based on the project.

Since the mounting orientation of the board also can be changed,
it is usefull to have an extended support for 180 degree rotation,
that can be defined for each individual projects.

Signed-off-by: rbudai98 <robert.budai@analog.com>

rbudai98 committed:
Eval-PowRMS: extend aprox range 5, 5.25, 5.5, 5.75, and 6 GHz

Signed-off-by: Robert Budai <robert.budai@analog.com>
